pkg/remotemetrics/matcher_test.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remotemetrics
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+type trueMatcher struct{}
+
+func (m *trueMatcher) Match(line string) bool {
+	return true
+}
+
+type falseMatcher struct{}
+
+func (m *falseMatcher) Match(line string) bool {
+	return false
+}
+
+var _ = Context("MatcherCollection", func() {
+
+	var matcher MatcherCollection
+	var line string
+	var res bool
+
+	JustBeforeEach(func() {
+		res = matcher.Match(line)
+	})
+
+	Context("All True", func() {
+
+		var matchers []Matcher
+
+		BeforeEach(func() {
+			matchers = []Matcher{
+				&trueMatcher{},
+				&trueMatcher{},
+			}
+		})
+
+		Context("matchAll", func() {
+
+			BeforeEach(func() {
+				matcher = MatchAll()
+
+				for _, m := range matchers {
+					matcher.Add(m)
+				}
+			})
+
+			It("should match", func() {
+				Expect(res).To(BeTrue())
+			})
```

Comment:
Personal taste, could compress to a single line to compact the test, given the lines are very short.